Additional scenarios in connection pool cleanup can trigger exception

**Describe the bug**
Removing streams not related to the connection for additional scenarios 

**To Reproduce**
```
org.agrona.concurrent.AgentTerminationException: java.lang.NullPointerException: Cannot invoke "io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientConnectionPool$KafkaClientStream.doStreamSignalNow(int)" because "stream" is null
        at io.aklivity.zilla.runtime.engine@0.9.53/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:681)
        at org.agrona.core/org.agrona.concurrent.AgentRunner.doDutyCycle(AgentRunner.java:291)
        at org.agrona.core/org.agrona.concurrent.AgentRunner.run(AgentRunner.java:164)
        at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: java.lang.NullPointerException: Cannot invoke "io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientConnectionPool$KafkaClientStream.doStreamSignalNow(int)" because "stream" is null
        at io.aklivity.zilla.runtime.binding.kafka@0.9.53/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientConnectionPool$KafkaClientSignaler.signalNow(KafkaClientConnectionPool.java:477)
        at io.aklivity.zilla.runtime.binding.kafka@0.9.53/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientGroupFactory$CoordinatorClient.doJoinGroupRequest(KafkaClientGroupFactory.java:3927)
        at io.aklivity.zilla.runtime.binding.kafka@0.9.53/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientGroupFactory$KafkaGroupStream.onApplicationMigrate(KafkaClientGroupFactory.java:1682)
        at io.aklivity.zilla.runtime.binding.kafka@0.9.53/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientGroupFactory.newStream(KafkaClientGroupFactory.java:368)
        at io.aklivity.zilla.runtime.binding.kafka@0.9.53/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientFactory.newStream(KafkaClientFactory.java:141)
        at io.aklivity.zilla.runtime.engine@0.9.53/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.handleBeginInitial(DispatchAgent.java:1350)
        at io.aklivity.zilla.runtime.engine@0.9.53/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.handleDefaultReadInitial(DispatchAgent.java:1118)
        at io.aklivity.zilla.runtime.engine@0.9.53/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.handleReadInitial(DispatchAgent.java:1058)
        at io.aklivity.zilla.runtime.engine@0.9.53/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.handleRead(DispatchAgent.java:1005)
        at io.aklivity.zilla.runtime.engine@0.9.53/io.aklivity.zilla.runtime.engine.internal.concurent.ManyToOneRingBuffer.read(ManyToOneRingBuffer.java:181)
        at io.aklivity.zilla.runtime.engine@0.9.53/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:675)
        ... 3 more
        Suppressed: java.lang.Exception: [engine/data#7]        [0x07070000000003ad] streams=[consumeAt=0x000926c0 (0x00000000000926c0), produceAt=0x000927a0 (0x00000000000927a0)]
                at io.aklivity.zilla.runtime.engine@0.9.53/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:679)
                ... 3 more
java.lang.NullPointerException: Cannot invoke "io.aklivity.zilla.runtime.engine.poller.PollerKey.clear(int)" because "this.networkKey" is null
        at io.aklivity.zilla.runtime.binding.tcp@0.9.53/io.aklivity.zilla.runtime.binding.tcp.internal.stream.TcpClientFactory$TcpClient.doNetShutdownOutput(TcpClientFactory.java:469)
        at io.aklivity.zilla.runtime.binding.tcp@0.9.53/io.aklivity.zilla.runtime.binding.tcp.internal.stream.TcpClientFactory$TcpClient.onAppAbort(TcpClientFactory.java:642)
        at io.aklivity.zilla.runtime.binding.tcp@0.9.53/io.aklivity.zilla.runtime.binding.tcp.internal.stream.TcpClientFactory$TcpClient.onAppMessage(TcpClientFactory.java:516)
        at io.aklivity.zilla.runtime.binding.tcp@0.9.53/io.aklivity.zilla.runtime.binding.tcp.internal.stream.TcpClientFactory.lambda$newStream$0(TcpClientFactory.java:166)
        at io.aklivity.zilla.runtime.engine@0.9.53/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doSyntheticAbort(DispatchAgent.java:1457)
        at io.aklivity.zilla.runtime.engine@0.9.53/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.lambda$detachStreams$0(DispatchAgent.java:499)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at io.aklivity.zilla.runtime.engine@0.9.53/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.detachStreams(DispatchAgent.java:494)
        at io.aklivity.zilla.runtime.engine@0.9.53/io.aklivity.zilla.runtime.engine.internal.registry.NamespaceRegistry.detachBinding(NamespaceRegistry.java:240)
        at java.base/java.util.Arrays$ArrayList.forEach(Arrays.java:4204)
        at io.aklivity.zilla.runtime.engine@0.9.53/io.aklivity.zilla.runtime.engine.internal.registry.NamespaceRegistry.detach(NamespaceRegistry.java:127)
        at io.aklivity.zilla.runtime.engine@0.9.53/io.aklivity.zilla.runtime.engine.internal.registry.ConfigurationRegistry.lambda$detachAll$0(ConfigurationRegistry.java:141)
stopped
        at org.agrona.core/org.agrona.collections.Int2ObjectHashMap$ValueCollection.forEach(Int2ObjectHashMap.java:747)
        at io.aklivity.zilla.runtime.engine@0.9.53/io.aklivity.zilla.runtime.engine.internal.registry.ConfigurationRegistry.detachAll(ConfigurationRegistry.java:141)
        at io.aklivity.zilla.runtime.engine@0.9.53/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.onClose(DispatchAgent.java:701)
        at org.agrona.core/org.agrona.concurrent.AgentRunner.run(AgentRunner.java:169)
        at java.base/java.lang.Thread.run(Thread.java:1623)
```

--- a/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientConnectionPool.java
+++ b/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientConnectionPool.java
@@ -1198,7 +1198,7 @@ public final class KafkaClientConnectionPool
 
             doConnectionEnd(traceId);
 
-            streamsByInitialIds.forEach((k, v) -> v.cleanupStream(traceId));
+            cleanupStreams(traceId);
         }
 
         private void onConnectionAbort(
@@ -1208,6 +1208,12 @@ public final class KafkaClientConnectionPool
 
             doConnectionAbort(traceId);
 
+            cleanupStreams(traceId);
+        }
+
+        private void cleanupStreams(
+            long traceId)
+        {
             streamsByInitialIds.forEach((k, v) ->
             {
                 if (v.connection == this)
@@ -1246,7 +1252,7 @@ public final class KafkaClientConnectionPool
 
             doConnectionReset(traceId);
 
-            streamsByInitialIds.forEach((k, v) -> v.cleanupStream(traceId));
+            cleanupStreams(traceId);
         }
 
         private void onConnectionWindow(

--- a/runtime/binding-kafka/src/test/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/ClientGroupIT.java
+++ b/runtime/binding-kafka/src/test/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/ClientGroupIT.java
@@ -70,6 +70,26 @@ public class ClientGroupIT
     @Test
     @Configuration("client.yaml")
     @Specification({
+        "${app}/client.sent.write.abort.after.sync.group.response/client",
+        "${net}/client.sent.read.abort.after.sync.group.response/server"})
+    public void shouldHandleClientSentReadAbortAfterSyncGroupResponse() throws Exception
+    {
+        k3po.finish();
+    }
+
+    @Test
+    @Configuration("client.yaml")
+    @Specification({
+        "${app}/client.sent.write.abort.after.sync.group.response/client",
+        "${net}/client.sent.write.close.after.sync.group.response/server"})
+    public void shouldHandleClientSentWriteCloseAfterSyncGroupResponse() throws Exception
+    {
+        k3po.finish();
+    }
+
+    @Test
+    @Configuration("client.yaml")
+    @Specification({
         "${app}/rebalance.protocol.highlander/client",
         "${net}/rebalance.protocol.highlander/server"})
     public void shouldLeaveGroupOnGroupRebalanceError() throws Exception

--- a/specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/network/group.f1.j5.s3.l3.h3/client.sent.read.abort.after.sync.group.response/client.rpt
+++ b/specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/network/group.f1.j5.s3.l3.h3/client.sent.read.abort.after.sync.group.response/client.rpt
@@ -1,0 +1,173 @@
+#
+# Copyright 2021-2023 Aklivity Inc.
+#
+# Aklivity licenses this file to you under the Apache License,
+# version 2.0 (the "License"); you may not use this file except in compliance
+# with the License. You may obtain a copy of the License at:
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+property networkConnectWindow 8192
+property instanceId ${kafka:randomBytes(42)}
+
+property newRequestId ${kafka:newRequestId()}
+property fetchWaitMax 500
+property fetchBytesMax 65535
+property partitionBytesMax 8192
+
+connect "zilla://streams/net0"
+  option zilla:window ${networkConnectWindow}
+  option zilla:transmission "duplex"
+  option zilla:byteorder "network"
+
+connected
+
+write  22                                # size
+       10s                               # find coordinator
+       1s                                # v1
+       ${newRequestId}
+       5s "zilla"                        # client id
+       4s "test"                         # "session" coordinator key
+       [0x00]                            # coordinator group type
+
+read  35                                 # size
+      (int:newRequestId)
+      0                                  # throttle time
+      0s                                 # no error
+      4s "none"                          # error message none
+      0                                  # coordinator node
+      9s "localhost"                     # host
+      9092                               # port
+
+write 82                                                 # size
+      32s                                                # describe configs
+      0s                                                 # v0
+      ${newRequestId}
+      -1s                                                # no client id
+      1                                                  # resources
+        [0x04]                                           # broker resource
+        1s "0"                                           # "node" topic
+        2                                                # configs
+          28s "group.min.session.timeout.ms"             # name
+          28s "group.max.session.timeout.ms"             # name
+
+read 103                                                # size
+      (int:newRequestId)
+      0
+      1                                                  # resources
+        0s                                               # no error
+        -1s                                              # error message
+        [0x04]                                           # broker resource
+        1s "0"                                           # "0" nodeId
+        2                                                # configs
+          28s "group.min.session.timeout.ms"             # name
+          4s "6000"                                      # value
+          [0x00]                                          # not read only
+          [0x00]                                          # not default
+          [0x00]                                          # not sensitive
+          28s "group.max.session.timeout.ms"             # name
+          5s "30000"                                     # value
+          [0x00]                                          # not read only
+          [0x00]                                          # not default
+          [0x00]                                          # not sensitive
+
+read aborted
+write abort
+
+read notify ROUTED_CLUSTER_SERVER
+
+connect await ROUTED_CLUSTER_SERVER
+        "zilla://streams/net0"
+  option zilla:window ${networkConnectWindow}
+  option zilla:transmission "duplex"
+  option zilla:byteorder "network"
+
+connected
+
+write 119                               # size
+      11s                               # join group
+      5s                                # v5
+      ${newRequestId}
+      5s "zilla"                        # client id
+      4s "test"                         # consumer group
+      30000                             # session timeout
+      4000                              # rebalance timeout
+      0s                                # consumer group member
+      42s ${instanceId}                 # group instance id
+      8s "consumer"                     # protocol type
+      1                                 # group protocol
+        10s "highlander"                     # protocol name
+        14                                   # metadata size
+        ${kafka:randomBytes(14)}             # metadata
+
+read  34                                                # size
+      (int:newRequestId)
+      0                                                 # throttle time
+      79s                                               # member id required
+      -1                                                # generated id
+      0s                                                # protocol name
+      0s                                                # leader id
+      10s "memberId-1"                                  # consumer member group id
+      0                                                 # members
+
+write 129                               # size
+      11s                               # join group
+      5s                                # v5
+      ${newRequestId}
+      5s "zilla"                        # no client id
+      4s "test"                         # consumer group
+      30000                             # session timeout
+      4000                              # rebalance timeout
+      10s "memberId-1"                  # consumer group member
+      42s ${instanceId}                 # group instance id
+      8s "consumer"                     # protocol type
+      1                                 # group protocol
+        10s "highlander"                     # protocol name
+        14                                   # metadata size
+        ${kafka:randomBytes(14)}             # metadata
+
+read  128                                               # size
+      (int:newRequestId)
+      0                                                 # throttle time
+      0s                                                # no error
+      3                                                 # generated id
+      10s "highlander"                                  # protocol name
+      10s "memberId-1"                                  # leader id
+      10s "memberId-1"                                  # consumer member group id
+      1                                                 # members
+         10s "memberId-1"                                  # consumer member group id
+         42s [0..42]                                       # group instance id
+         14                                                # metadata size
+         2s                                                # version
+         0                                                 # topics
+         0                                                 # userdata
+         0                                                 # partitions
+
+write 101                                            # size
+      14s                                           # sync group
+      3s                                            # v3
+      ${newRequestId}
+      5s "zilla"                                    # no client id
+      4s "test"                                     # consumer group
+      3                                             # generation id
+      10s "memberId-1"                              # consumer member group id
+      42s ${instanceId}                             # group instance id
+      1                                             # assignments
+        10s "memberId-1"                             # consumer member group id
+        0                                            # metadata
+
+read  14                                                # size
+      (int:newRequestId)
+      0                                                 # throttle time
+      0s                                                # no error
+      0                                                 # assignment
+
+write aborted
+read abort

--- a/specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/network/group.f1.j5.s3.l3.h3/client.sent.read.abort.after.sync.group.response/server.rpt
+++ b/specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/network/group.f1.j5.s3.l3.h3/client.sent.read.abort.after.sync.group.response/server.rpt
@@ -1,0 +1,164 @@
+#
+# Copyright 2021-2023 Aklivity Inc.
+#
+# Aklivity licenses this file to you under the Apache License,
+# version 2.0 (the "License"); you may not use this file except in compliance
+# with the License. You may obtain a copy of the License at:
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+property instanceId ${kafka:randomBytes(42)}
+
+accept "zilla://streams/net0"
+  option zilla:window 8192
+  option zilla:transmission "duplex"
+  option zilla:byteorder "network"
+
+accepted
+
+connected
+
+read  22                                # size
+      10s                               # find coordinator
+      1s                                # v1
+      (int:newRequestId)
+      5s "zilla"                        # client id
+      4s "test"                         # "test" coordinator key
+      [0x00]                            # coordinator group type
+
+write 35                                 # size
+      ${newRequestId}
+      0                                  # throttle time
+      0s                                 # no error
+      4s "none"                          # error message none
+      0                                  # coordinator node
+      9s "localhost"                  # host
+      9092                            # port
+
+read 82                                                 # size
+     32s                                                # describe configs
+     0s                                                 # v0
+     (int:requestId)
+     -1s                                                # no client id
+     1                                                  # resources
+       [0x04]                                           # broker resource
+       1s "0"                                           # "node" topic
+       2                                                # configs
+         28s "group.min.session.timeout.ms"             # name
+         28s "group.max.session.timeout.ms"             # name
+
+write 103                                                # size
+      ${requestId}
+      0
+      1                                                  # resources
+        0s                                               # no error
+        -1s                                              # error message
+        [0x04]                                           # broker resource
+        1s "0"                                           # "0" nodeId
+        2                                                # configs
+          28s "group.min.session.timeout.ms"             # name
+          4s "6000"                                      # value
+          [0x00]                                          # not read only
+          [0x00]                                          # not default
+          [0x00]                                          # not sensitive
+          28s "group.max.session.timeout.ms"             # name
+          5s "30000"                                     # value
+          [0x00]                                          # not read only
+          [0x00]                                          # not default
+          [0x00]                                          # not sensitive
+
+write abort
+read aborted
+
+accepted
+
+connected
+
+read  119                               # size
+      11s                               # join group
+      5s                                # v5
+      (int:newRequestId)
+      5s "zilla"                        # client id
+      4s "test"                         # consumer group
+      30000                             # session timeout
+      4000                              # rebalance timeout
+      0s                                # consumer group member
+      42s [0..42]                       # group instance id
+      8s "consumer"                     # protocol type
+      1                                 # group protocol
+        10s "highlander"                     # protocol name
+        14                                   # metadata size
+        [0..14]                              # metadata
+
+
+write 34                                                # size
+      ${newRequestId}
+      0                                                 # throttle time
+      79s                                               # member id required
+      -1                                                # generated id
+      0s                                                # protocol name
+      0s                                                # leader id
+      10s "memberId-1"                                  # consumer member group id
+      0                                                 # members
+
+read  129                               # size
+      11s                               # join group
+      5s                                # v5
+      (int:newRequestId)
+      5s "zilla"                        # no client id
+      4s "test"                         # consumer group
+      30000                             # session timeout
+      4000                              # rebalance timeout
+      10s "memberId-1"                  # consumer group member
+      42s [0..42]                       # group instance id
+      8s "consumer"                     # protocol type
+      1                                 # group protocol
+        10s "highlander"                    # protocol name
+        14                                   # metadata size
+        [0..14]                              # metadata
+
+write 128                                               # size
+      ${newRequestId}
+      0                                                 # throttle time
+      0s                                                # no error
+      3                                                 # generated id
+      10s "highlander"                                  # protocol name
+      10s "memberId-1"                                  # leader id
+      10s "memberId-1"                                  # consumer member group id
+      1                                                 # members
+         10s "memberId-1"                                  # consumer member group id
+         42s ${instanceId}                                 # group instance id
+         14                                                # metadata size
+         2s                                                # version
+         0                                                 # topics
+         0                                                 # userdata
+         0                                                 # partitions
+
+read 101                                           # size
+     14s                                           # sync group
+     3s                                            # v3
+     (int:newRequestId)
+     5s "zilla"                                    # no client id
+     4s "test"                                     # consumer group
+     3                                             # generation id
+     10s "memberId-1"                              # consumer member group id
+     42s [0..42]                                   # group instance id
+     1                                             # assignments
+       10s "memberId-1"                             # consumer member group id
+       0                                            # metadata
+
+write 14                                                # size
+      ${newRequestId}
+      0                                                 # throttle time
+      0s                                                # no error
+      0                                                 # assignment
+
+read abort
+write aborted

--- a/specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/network/group.f1.j5.s3.l3.h3/client.sent.write.close.after.sync.group.response/client.rpt
+++ b/specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/network/group.f1.j5.s3.l3.h3/client.sent.write.close.after.sync.group.response/client.rpt
@@ -1,0 +1,173 @@
+#
+# Copyright 2021-2023 Aklivity Inc.
+#
+# Aklivity licenses this file to you under the Apache License,
+# version 2.0 (the "License"); you may not use this file except in compliance
+# with the License. You may obtain a copy of the License at:
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+property networkConnectWindow 8192
+property instanceId ${kafka:randomBytes(42)}
+
+property newRequestId ${kafka:newRequestId()}
+property fetchWaitMax 500
+property fetchBytesMax 65535
+property partitionBytesMax 8192
+
+connect "zilla://streams/net0"
+  option zilla:window ${networkConnectWindow}
+  option zilla:transmission "duplex"
+  option zilla:byteorder "network"
+
+connected
+
+write  22                                # size
+       10s                               # find coordinator
+       1s                                # v1
+       ${newRequestId}
+       5s "zilla"                        # client id
+       4s "test"                         # "session" coordinator key
+       [0x00]                            # coordinator group type
+
+read  35                                 # size
+      (int:newRequestId)
+      0                                  # throttle time
+      0s                                 # no error
+      4s "none"                          # error message none
+      0                                  # coordinator node
+      9s "localhost"                     # host
+      9092                               # port
+
+write 82                                                 # size
+      32s                                                # describe configs
+      0s                                                 # v0
+      ${newRequestId}
+      -1s                                                # no client id
+      1                                                  # resources
+        [0x04]                                           # broker resource
+        1s "0"                                           # "node" topic
+        2                                                # configs
+          28s "group.min.session.timeout.ms"             # name
+          28s "group.max.session.timeout.ms"             # name
+
+read 103                                                # size
+      (int:newRequestId)
+      0
+      1                                                  # resources
+        0s                                               # no error
+        -1s                                              # error message
+        [0x04]                                           # broker resource
+        1s "0"                                           # "0" nodeId
+        2                                                # configs
+          28s "group.min.session.timeout.ms"             # name
+          4s "6000"                                      # value
+          [0x00]                                          # not read only
+          [0x00]                                          # not default
+          [0x00]                                          # not sensitive
+          28s "group.max.session.timeout.ms"             # name
+          5s "30000"                                     # value
+          [0x00]                                          # not read only
+          [0x00]                                          # not default
+          [0x00]                                          # not sensitive
+
+read aborted
+write abort
+
+read notify ROUTED_CLUSTER_SERVER
+
+connect await ROUTED_CLUSTER_SERVER
+        "zilla://streams/net0"
+  option zilla:window ${networkConnectWindow}
+  option zilla:transmission "duplex"
+  option zilla:byteorder "network"
+
+connected
+
+write 119                               # size
+      11s                               # join group
+      5s                                # v5
+      ${newRequestId}
+      5s "zilla"                        # client id
+      4s "test"                         # consumer group
+      30000                             # session timeout
+      4000                              # rebalance timeout
+      0s                                # consumer group member
+      42s ${instanceId}                 # group instance id
+      8s "consumer"                     # protocol type
+      1                                 # group protocol
+        10s "highlander"                     # protocol name
+        14                                   # metadata size
+        ${kafka:randomBytes(14)}             # metadata
+
+read  34                                                # size
+      (int:newRequestId)
+      0                                                 # throttle time
+      79s                                               # member id required
+      -1                                                # generated id
+      0s                                                # protocol name
+      0s                                                # leader id
+      10s "memberId-1"                                  # consumer member group id
+      0                                                 # members
+
+write 129                               # size
+      11s                               # join group
+      5s                                # v5
+      ${newRequestId}
+      5s "zilla"                        # no client id
+      4s "test"                         # consumer group
+      30000                             # session timeout
+      4000                              # rebalance timeout
+      10s "memberId-1"                  # consumer group member
+      42s ${instanceId}                 # group instance id
+      8s "consumer"                     # protocol type
+      1                                 # group protocol
+        10s "highlander"                     # protocol name
+        14                                   # metadata size
+        ${kafka:randomBytes(14)}             # metadata
+
+read  128                                               # size
+      (int:newRequestId)
+      0                                                 # throttle time
+      0s                                                # no error
+      3                                                 # generated id
+      10s "highlander"                                  # protocol name
+      10s "memberId-1"                                  # leader id
+      10s "memberId-1"                                  # consumer member group id
+      1                                                 # members
+         10s "memberId-1"                                  # consumer member group id
+         42s [0..42]                                       # group instance id
+         14                                                # metadata size
+         2s                                                # version
+         0                                                 # topics
+         0                                                 # userdata
+         0                                                 # partitions
+
+write 101                                            # size
+      14s                                           # sync group
+      3s                                            # v3
+      ${newRequestId}
+      5s "zilla"                                    # no client id
+      4s "test"                                     # consumer group
+      3                                             # generation id
+      10s "memberId-1"                              # consumer member group id
+      42s ${instanceId}                             # group instance id
+      1                                             # assignments
+        10s "memberId-1"                             # consumer member group id
+        0                                            # metadata
+
+read  14                                                # size
+      (int:newRequestId)
+      0                                                 # throttle time
+      0s                                                # no error
+      0                                                 # assignment
+
+read closed
+write close

--- a/specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/network/group.f1.j5.s3.l3.h3/client.sent.write.close.after.sync.group.response/server.rpt
+++ b/specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/network/group.f1.j5.s3.l3.h3/client.sent.write.close.after.sync.group.response/server.rpt
@@ -1,0 +1,164 @@
+#
+# Copyright 2021-2023 Aklivity Inc.
+#
+# Aklivity licenses this file to you under the Apache License,
+# version 2.0 (the "License"); you may not use this file except in compliance
+# with the License. You may obtain a copy of the License at:
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+property instanceId ${kafka:randomBytes(42)}
+
+accept "zilla://streams/net0"
+  option zilla:window 8192
+  option zilla:transmission "duplex"
+  option zilla:byteorder "network"
+
+accepted
+
+connected
+
+read  22                                # size
+      10s                               # find coordinator
+      1s                                # v1
+      (int:newRequestId)
+      5s "zilla"                        # client id
+      4s "test"                         # "test" coordinator key
+      [0x00]                            # coordinator group type
+
+write 35                                 # size
+      ${newRequestId}
+      0                                  # throttle time
+      0s                                 # no error
+      4s "none"                          # error message none
+      0                                  # coordinator node
+      9s "localhost"                  # host
+      9092                            # port
+
+read 82                                                 # size
+     32s                                                # describe configs
+     0s                                                 # v0
+     (int:requestId)
+     -1s                                                # no client id
+     1                                                  # resources
+       [0x04]                                           # broker resource
+       1s "0"                                           # "node" topic
+       2                                                # configs
+         28s "group.min.session.timeout.ms"             # name
+         28s "group.max.session.timeout.ms"             # name
+
+write 103                                                # size
+      ${requestId}
+      0
+      1                                                  # resources
+        0s                                               # no error
+        -1s                                              # error message
+        [0x04]                                           # broker resource
+        1s "0"                                           # "0" nodeId
+        2                                                # configs
+          28s "group.min.session.timeout.ms"             # name
+          4s "6000"                                      # value
+          [0x00]                                          # not read only
+          [0x00]                                          # not default
+          [0x00]                                          # not sensitive
+          28s "group.max.session.timeout.ms"             # name
+          5s "30000"                                     # value
+          [0x00]                                          # not read only
+          [0x00]                                          # not default
+          [0x00]                                          # not sensitive
+
+write abort
+read aborted
+
+accepted
+
+connected
+
+read  119                               # size
+      11s                               # join group
+      5s                                # v5
+      (int:newRequestId)
+      5s "zilla"                        # client id
+      4s "test"                         # consumer group
+      30000                             # session timeout
+      4000                              # rebalance timeout
+      0s                                # consumer group member
+      42s [0..42]                       # group instance id
+      8s "consumer"                     # protocol type
+      1                                 # group protocol
+        10s "highlander"                     # protocol name
+        14                                   # metadata size
+        [0..14]                              # metadata
+
+
+write 34                                                # size
+      ${newRequestId}
+      0                                                 # throttle time
+      79s                                               # member id required
+      -1                                                # generated id
+      0s                                                # protocol name
+      0s                                                # leader id
+      10s "memberId-1"                                  # consumer member group id
+      0                                                 # members
+
+read  129                               # size
+      11s                               # join group
+      5s                                # v5
+      (int:newRequestId)
+      5s "zilla"                        # no client id
+      4s "test"                         # consumer group
+      30000                             # session timeout
+      4000                              # rebalance timeout
+      10s "memberId-1"                  # consumer group member
+      42s [0..42]                       # group instance id
+      8s "consumer"                     # protocol type
+      1                                 # group protocol
+        10s "highlander"                    # protocol name
+        14                                   # metadata size
+        [0..14]                              # metadata
+
+write 128                                               # size
+      ${newRequestId}
+      0                                                 # throttle time
+      0s                                                # no error
+      3                                                 # generated id
+      10s "highlander"                                  # protocol name
+      10s "memberId-1"                                  # leader id
+      10s "memberId-1"                                  # consumer member group id
+      1                                                 # members
+         10s "memberId-1"                                  # consumer member group id
+         42s ${instanceId}                                 # group instance id
+         14                                                # metadata size
+         2s                                                # version
+         0                                                 # topics
+         0                                                 # userdata
+         0                                                 # partitions
+
+read 101                                           # size
+     14s                                           # sync group
+     3s                                            # v3
+     (int:newRequestId)
+     5s "zilla"                                    # no client id
+     4s "test"                                     # consumer group
+     3                                             # generation id
+     10s "memberId-1"                              # consumer member group id
+     42s [0..42]                                   # group instance id
+     1                                             # assignments
+       10s "memberId-1"                             # consumer member group id
+       0                                            # metadata
+
+write 14                                                # size
+      ${newRequestId}
+      0                                                 # throttle time
+      0s                                                # no error
+      0                                                 # assignment
+
+write close
+read closed

--- a/specs/binding-kafka.spec/src/test/java/io/aklivity/zilla/specs/binding/kafka/streams/network/GroupIT.java
+++ b/specs/binding-kafka.spec/src/test/java/io/aklivity/zilla/specs/binding/kafka/streams/network/GroupIT.java
@@ -65,6 +65,24 @@ public class GroupIT
 
     @Test
     @Specification({
+        "${net}/client.sent.read.abort.after.sync.group.response/client",
+        "${net}/client.sent.read.abort.after.sync.group.response/server"})
+    public void shouldHandleClientSentReadAbortAfterSyncGroupResponse() throws Exception
+    {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
+        "${net}/client.sent.write.close.after.sync.group.response/client",
+        "${net}/client.sent.write.close.after.sync.group.response/server"})
+    public void shouldHandleClientSentWriteClosetAfterSyncGroupResponse() throws Exception
+    {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
         "${net}/coordinator.not.available/client",
         "${net}/coordinator.not.available/server"})
     public void shouldHandleCoordinatorNotAvailableError() throws Exception

